src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -897,34 +946,37 @@ WORD Terminal::_TakeVirtualKeyFromLastKeyEvent(const WORD scanCode) noexcept
 
 Viewport Terminal::_GetMutableViewport() const noexcept
 {
-    return _mutableViewport;
+    return _inAltBuffer() ? Viewport::FromDimensions(_mutableViewport.Dimensions()) :
+                            _mutableViewport;
 }
 
 short Terminal::GetBufferHeight() const noexcept
 {
-    return _mutableViewport.BottomExclusive();
+    return _GetMutableViewport().BottomExclusive();
 }
 
 // ViewStartIndex is also the length of the scrollback
 int Terminal::ViewStartIndex() const noexcept
 {
-    return _mutableViewport.Top();
+    return _inAltBuffer() ? 0 : _mutableViewport.Top();
 }
 
 int Terminal::ViewEndIndex() const noexcept
 {
-    return _mutableViewport.BottomInclusive();
+    return _inAltBuffer() ? _mutableViewport.Height() - 1 : _mutableViewport.BottomInclusive();
 }
 
 // _VisibleStartIndex is the first visible line of the buffer
 int Terminal::_VisibleStartIndex() const noexcept
 {
-    return std::max(0, ViewStartIndex() - _scrollOffset);
+    return _inAltBuffer() ? ViewStartIndex() :
+                            std::max(0, ViewStartIndex() - _scrollOffset);
 }
 
 int Terminal::_VisibleEndIndex() const noexcept
 {
-    return std::max(0, ViewEndIndex() - _scrollOffset);
+    return _inAltBuffer() ? ViewEndIndex() :
+                            std::max(0, ViewEndIndex() - _scrollOffset);
 }

```

Comment:
I don't think these changes are necessary. I believe it'd be better to leave the code as it is, because it'd better to get rid of the `Viewport` class altogether which is easier with the previous code.

Also since these methods might be used for clamping values, I believe that this might introduce bugs more easily.

Reply:
I am guessing they are necessary because of `_scrollOffset`... which makes me wonder how we account for the scroll offset being per-buffer but global state

Reply:
We could use a getter for `_scrollOffset` which returns 0 for the alt buffer. 🤔